test/testData/read/mapping.ts

```diff
@@ -52,6 +52,12 @@ export function mapGyroPoolStateToPrismaPool(
         });
     });
 
+    // transform hook dynamicData values to bigInt and then apply formatEther to them
+    const _hookDynamicData = poolState.hook?.dynamicData;
+    const hookDynamicData = _hookDynamicData
+        ? Object.fromEntries(Object.entries(_hookDynamicData).map(([key, value]) => [key, formatEther(BigInt(value))]))
+        : undefined;
+
```

Comment:
API <-> SOR/SDK scaling is often an uncertainty for me. Is it always the case that dynamic data is scaled via 1e18?

Reply:
It's a good point. Balancer maths will always expect the scaling to be same as whats used onchain. I think it has always been 1e18 scaled so far but might not always be the case. Probably need to be aware of that as a potential issue and handle on a case by case basic. WDYT @brunoguerios ?

Reply:
Yeah, that's a valid concern. 
I set it to 1e18 because all existing hook parameters use that scaling so far. 
If any hook ever uses a different scale, it will have to provide that extra bit of information so we can parse it accordingly on the backend (which uses float point).
Since we'll have to add support for each new hook type individually, I'd say it's ok to handle it when it becomes an issue.
Ps: since fetching on-chain hook data is performed by the backend itself (and we reuse the code), whenever such issue exists, by the time it reaches the SOR, it will likely be already solved 🙃